File: src/main/java/no/fint/TemplateService.java
package no.fint;

import org.jooq.lambda.tuple.Tuple2;
import org.springframework.stereotype.Service;
import org.thymeleaf.TemplateEngine;
import org.thymeleaf.context.Context;
import no.fint.organization.OrganizationDocument;

import java.util.List;

@Service
public class TemplateService {

    private final TemplateEngine templateEngine;

    public TemplateService(TemplateEngine templateEngine) {
        this.templateEngine = templateEngine;
    }

    public String render(List<OrganizationDocument> added, List<Tuple2<OrganizationDocument, OrganizationDocument>> updated) {
        Context context = new Context();
        context.setVariable("added", added);
        context.setVariable("updated", updated);
        return templateEngine.process("email-template", context);
    }
}

File: src/main/kotlin/no/fint/orgmonitor/utils/TemplateService.kt
package no.fint.orgmonitor.utils

import no.fint.orgmonitor.organization.OrganizationDocument
import no.fint.orgmonitor.organization.SimpleOrganizationInfo
import org.springframework.stereotype.Service
import org.thymeleaf.TemplateEngine
import org.thymeleaf.context.Context

@Service
class TemplateService(
    private val templateEngine: TemplateEngine,
) {
    fun render(
        added: List<OrganizationDocument>,
        updated: List<Pair<OrganizationDocument, OrganizationDocument>>,
        parentInfo: List<SimpleOrganizationInfo>,
    ): String =
        templateEngine.process(
            "email-template",
            Context().apply {
                setVariable("added", added)
                setVariable("updated", updated)
                setVariable("parentInfo", parentInfo)
            },
        )
}
